2.13/tests/test/test.js

```diff
@@ -497,57 +489,69 @@ function testAuthentication(handler, verb) {
                     .expect(401, done);
             }
         })
-    }    
+    }
 }
 
 function testAPIVersionHeader(handler, verb) {
     it('should reject requests without X-Broker-API-Version header with 412', function(done) {
         if (verb == 'GET') {
             preparedRequest()
-                .get(handler)                    
+                .get(handler)
                 .auth(config.user, config.password)
                 .expect(412, done)
         } else if (verb == 'PUT') {
             preparedRequest()
-                .put(handler)                    
+                .put(handler)
                 .auth(config.user, config.password)
                 .send({})
                 .expect(412, done)
         } else if (verb == 'PATCH') {
             preparedRequest()
-                .patch(handler)                    
+                .patch(handler)
                 .auth(config.user, config.password)
                 .send({})
                 .expect(412, done)
         } else if (verb == 'DELETE') {
             preparedRequest()
-                .delete(handler)                    
+                .delete(handler)
                 .auth(config.user, config.password)
                 .expect(412, done)
         }
-    })     
+    })
 }
-function testAsyncParameter(handler, verb) {
+
+function testAsyncParameter(handler, verb, body) {
     it ('should return 422 if request doesn\'t have the accepts_incomplete parameter', function(done) {

```

Comment:
What is the thinking behind assuming the broker will fail this request? A broker that supports async doesn't have to require it.

Reply:
We can't find in the spec where it says it's only required when the broker in a synchronous broker. 

For example, on the [Update body](https://github.com/openservicebrokerapi/servicebroker/blob/v2.13/spec.md#body-4) spec, it says that `service_id` is required and our broker will check the presence of [this field](https://github.com/pivotal-cf/brokerapi/blob/b3e8b6720ab2c7f3383dc74251b20bf345736137/api.go#L225-L239) before checking the `accepts_incomplete` flag.

Reply:
The accepts_incomplete flag indicates that the broker CAN support async - not that it ONLY supports async. So, its possible that a broker includes it on all of its services even if it can respond very quickly to requests just so that it can work with Platforms that prefer an async processing model.

Checking for service_id before checking for accepts_incomplete is a valid choice, but its not mandated. All that's required is that all incoming fields are checked before any processing takes place. The spec doesn't mandate the order in which they are checked.

Reply:
Absolutely. I see where you are coming from and I agree; maybe this tests shouldn't exist in the first place. We think this discussion is outside the scope of this PR though, as we didn't introduce the `testAsyncParameter` function. What we did was to add the required body when sending the requests to the broker in order to isolate the behaviour being tested.

Reply:
Agree with @kirederik, this looks to be an existing construct. Let's merge the PR and raise an issue.